app/src/main/java/ch/blinkenlights/android/vanilla/SongTimeline.java

```diff
@@ -793,6 +860,14 @@ public int addSongs(Context context, QueryTask query)
 			case MODE_PLAY:
 			case MODE_PLAY_POS_FIRST:
 			case MODE_PLAY_ID_FIRST:
+				/*
```

Comment:
I think it would be better to generalize this and take a snapshot every time the queue is modified: this would also help in cases where users accidental clear the whole queue.

Maybe this could be hooked up into `saveActiveSongs()`

Reply:
We could. However, as queues get longer, we would get longer and longer snapshots in the memory. The main point of this feature is to undo massive edits. It is relatively easy to change one song. Rebuilding an entire queue is harder. With this, I feel that a little moderation would be needed, especially with memory concerns.